### PR TITLE
David's Pull request

### PR DESCRIPTION
**Pull request by dawi118**

### CODE REVIEW

**Review by sjmog — COMMENTED** [review]
Looks like you still have an extra file in here!

**Review by sjmog — COMMENTED** [review]
Passes exercise 1.1

**Review by sjmog — COMMENTED** [review]
Nice! Completes 1.2

File: calculator.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+def add(num=0)
+num
```
Comment (sjmog) [inline calculator.rb]: Please indent this correctly

File: calculator.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+def add(num=0)
```
Comment (sjmog) [inline calculator.rb]: `num` is not a very clear variable name. Consider using `number` instead.

**Review by sjmog — COMMENTED** [review]
Solves 1.3

File: calculator.rb [hunk 3]
```diff
@@ -1,3 +1,8 @@
+#Constants defined
```
Comment (sjmog) [inline calculator.rb]: I like that you separated this file with comments as it makes the program easier to read

File: calculator.rb [hunk 4]
```diff
@@ -1,3 +1,8 @@
...
+TWO = 2
+THREE = 3
+
+#Functions defined
 def add(num=0,num2=0)
```
Comment (sjmog) [inline calculator.rb]: Still not a fan of these variable names!

File: 2/greeter.rb [hunk 5]
```diff
@@ -1,2 +1,7 @@
-puts 'Hello, World!' do
+print 'Please enter your name:' do
+end
+
+name=gets.chomp
```
Comment (sjmog) [inline 2/greeter.rb:3]: How did you find out about this?

File: 2/greeter.rb [hunk 6]
```diff
@@ -1,2 +1,7 @@
-puts 'Hello, World!' do
+print 'Please enter your name:' do
```
Comment (sjmog) [inline 2/greeter.rb]: What are the `do` and `end` for here?

File: 2/greeter.rb [hunk 7]
```diff
@@ -1,2 +1,7 @@
...
+end
+
+name=gets.chomp
+
+puts 'Hello, ' + name + "!" do
```
Comment (sjmog) [inline 2/greeter.rb]: Ditto here – why the `do` and `end`?

File: 2/greeter.rb [hunk 8]
```diff
@@ -3,5 +3,5 @@
 
 name=gets.chomp
 
-puts 'Hello, ' + name + "!" do  
+puts 'Hello, ' + name.upcase + "!" do
```
Comment (sjmog) [inline 2/greeter.rb]: Nice! How did you find this?

File: 1/calculator.rb [hunk 9]
```diff
@@ -1,4 +1,33 @@
...
+EIGHT = 8
+NINE = 9
+
+#Functions defined
+
+def add(number=0,number2=0, number3=0,number4=0)
+  number + number2 + number3 + number4
+end
+
+def subtract(number=0,number_main=0)
```
Comment (sjmog) [inline 1/calculator.rb:19]: What was your thinking in using the name `number_main`? I like it, it distinguishes it from the other number.

File: 1/calculator.rb [hunk 10]
```diff
@@ -1,4 +1,33 @@
...
+
+def multiply(number=0, number2=0)
+  number * number2
+end
+
+def divide(number_main=0, number=0)
+  number_main / number
+end
+
 def hello
```
Comment (sjmog) [inline 1/calculator.rb]: Avoid committing unused code!

File: 2/greeter.rb (hunk 5 above)
Comment (dawi118) [reply 2/greeter.rb:3]: https://ruby-doc.org/docs/ruby-doc-bundle/Tutorial/part_02/user_input.html

Needed to stop the exclamation mark from being printed on a different line!

File: 2/greeter.rb (hunk 8 above)
Comment (dawi118) [reply 2/greeter.rb]: I remembered .upcase from the Makers' 'learn Ruby in 20 minutes' introduction

File: 1/calculator.rb (hunk 9 above)
Comment (dawi118) [reply 1/calculator.rb:19]: I had used `number_main` for the final number in subtraction which the other numbers subtracted from. Made sense to define it as the 'main number from which other numbers then interact with', which was also useful for division.

File: 2/greeter.rb (hunk 7 above)
Comment (dawi118) [reply 2/greeter.rb]: I had originally used 'do' as this allowed the programme to run, and so I thought that this was the solution. I can see it is not!

File: 2/greeter.rb (hunk 6 above)
Comment (dawi118) [reply 2/greeter.rb]: I had originally used 'do' as this allowed the programme to run, and so I thought that this was the solution. I can see it is not!

File: 1/tax_calculator.rb [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  income - take_home_pay(income,inc_tax_rate,stu_loan_rate,nat_ins_rate)
+end
+
+def fraction_of_people_who_will_pay(inc_tax_rate=0.1,stu_loan_rate=0.05,nat_ins_rate=0.05)
+  (1-(inc_tax_rate+stu_loan_rate+nat_ins_rate)).round(1)
+end
+
+def government_revenue(av_income=0, inc_tax_rate=0.1, stu_loan_rate=0.05, nat_ins_rate=0.05,pop_size=1000000)
+  (pop_size*fraction_of_people_who_will_pay(inc_tax_rate,stu_loan_rate,nat_ins_rate)*government_revenue_per_head(av_income,inc_tax_rate,stu_loan_rate,nat_ins_rate)).round(0)
+end
```
Comment (sjmog) [inline 1/tax_calculator.rb:35]: I like this!

**Pull request merged**